master/buildbot/__init__.py

```diff
@@ -80,6 +80,16 @@ def getVersion(init_file):
     except IOError:
         pass
 
+    git_archive_id = '$Format:%ct %d$'
+    if not re.search(r'^\$Format:', git_archive_id):
```

Comment:
I don't see how this test could fail. Are you sure you want to set git_archive_id just before?

Reply:
ahh. I downloaded the archive, and saw that git archive will actually replace that code!
We surely need to explain that with a comment in the code.

we can also rather say:
```
if not git_archive_id.startswith("$Format:"):
```

Reply:
This special `$Format:PLACEHOLDERS$` string is replaced by the expanded `PLACEHOLDERS` when building the archive (see https://git-scm.com/docs/gitattributes and https://git-scm.com/docs/pretty-formats).

If we are executing this code from the git repo, the export-subst thingy will not take place and `git_archive_id` **will** begin with `$Format:`. I only want to look into the contents of `git_archive_id` if the expansion took place.

Reply:
I can think of a unit test but that would require adding a tag and checking that git_archive_id is indeed modified after `git archive` by looking in the files.